zwrap/sample.go

```diff
@@ -98,70 +103,62 @@ func (s *sampler) With(fields ...zap.Field) zap.Logger {
 }
 
 func (s *sampler) Check(lvl zap.Level, msg string) *zap.CheckedMessage {
-	if !s.check(lvl, msg) {
-		return nil
+	cm := s.Logger.Check(lvl, msg)
+	switch lvl {
+	case zap.PanicLevel, zap.FatalLevel:
+		return cm
 	}
-	return zap.NewCheckedMessage(s.Logger, lvl, msg)
+	if s.sampled(msg) {
+		return cm
+	}
+	return nil
 }
 
 func (s *sampler) Log(lvl zap.Level, msg string, fields ...zap.Field) {
-	if s.check(lvl, msg) {
+	if s.Logger.Check(lvl, msg) != nil && s.sampled(msg) {
```

Comment:
Going to try going one step further on `cm.OK()`, and actually call `cm.Write` instead of `s.logger.Log`.